src/banner/Banner.jsx

```diff
@@ -0,0 +1,96 @@
+
+import BannerRegistrationContext from './private/BannerRegistrationContext';
+import { BANNER_TYPES } from './private/utils';
+
+/* eslint-disable react/no-unused-prop-types */
+const propTypes = {
+  /**
+   * An action element to be added to the action section of the banner.
+   */
+  action: PropTypes.element,
+  /**
+   * The pieces to populate a banner when type=`custom`.
+   */
+  custom: PropTypes.shape({
+    /**
+     * The title for the banner which will be bolded.
+     */
+    bannerTitle: PropTypes.string,
+    /**
+     * The class name used to set the icon in the banner.
+     */
+    icon: PropTypes.element,
+    /**
+     * Sets an author-defined class, to control the status bar color to be used.
+     */
+    colorClass: PropTypes.string,
+  }),
+  /**
+   * Nodes providing the message content for the banner. Can contain text and HTML.
+   */
+  description: PropTypes.node,
+  /**
+   * Callback function triggered when Dismiss button is clicked. The presence of this prop will cause
+   * the Dismiss button to be included on the banner.
+   */
+  onDismiss: PropTypes.func,
+  /**
+   * The type of alert to be rendered. One of `alert`, `error`, `warning`, `info`, `success`, `unsatisfied`, `unverified` or `custom`.
+   */
+  type: PropTypes.oneOf([
+    BANNER_TYPES.ALERT,
+    BANNER_TYPES.ERROR,
+    BANNER_TYPES.WARNING,
+    BANNER_TYPES.UNSATISFIED,
+    BANNER_TYPES.UNVERIFIED,
+    BANNER_TYPES.ADVISORY,
+    BANNER_TYPES.INFO,
+    BANNER_TYPES.SUCCESS,
+    BANNER_TYPES.CUSTOM,
+  ]).isRequired,
+};
+/* eslint-enable react/no-unused-prop-types */
+
+const Banner = (props) => {
+  const bannerRegistration = React.useContext(BannerRegistrationContext);
+
+  React.useEffect(() => {
+    /**
+     * A unique identifier is generated for each Banner during construction. This will be used to
+     * uniquely register/unregister the banner with ancestor Banner Managers without requiring consumers to
+     * define unique identifiers themselves.
+     */
+    const uuid = uuidv4();
+
+    /**
+     * If the bannerRegistration value is the ProviderRegistrationContext's default value,
+     * then there is not a matching BannerCheckpoint above it in the hierarchy.
+     * This is possible but likely not intentional, so the component warns.
+     */
+    if (!bannerRegistration && process.env.NODE_ENV !== 'production') {
+      /* eslint-disable no-console */
+      console.warn('A Banner was not rendered within the context of a BannerCheckpoint. If this is unexpected, validate that the expected version of the terra-application package is installed.');
+      /* eslint-enable no-console */
+      return () => {};
+    }
+
+    if (bannerRegistration && bannerRegistration.registerBanner) {
+      bannerRegistration.registerBanner(uuid, props);
+    }
+
+    return () => {
+      if (bannerRegistration && bannerRegistration.unregisterBanner) {
+        bannerRegistration.unregisterBanner(uuid, props.type);
```

Comment:
Having to unregister/re-register a banner with every prop change might lead to some flashing on slower devices if the updates aren't batched.

Have you considered only unregistering on component unmount (in a separate effect) and making registerBanner update an existing banner if one exists? With props like onDismiss that are typically provided with inline functions, this would happen quite often.

Reply:
I hadn't but that is a good idea. I went ahead an moved it out into it's own useEffect hook: https://github.com/cerner/terra-application/pull/45/commits/71e60344c4d90e82148ba7e8bf976d26c80f61b4